Add option to set macOS build version

Thomas-Vos opened the pull request:
Fixes #1611

Currently, the build version is set to the same version as the `packageVersion` property. This PR adds a new option, called `packageBuildVersion` which can be used to customise the build version. If this option is not specified, it will fallback to `packageVersion` for backwards compatibility.

olonho commented:
@AlexeyTsvetkov please review

AlexeyTsvetkov commented on gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractJPackageTask.kt:
```diff
@@ -477,7 +481,8 @@ abstract class AbstractJPackageTask @Inject constructor(
         plist[PlistKeys.CFBundleShortVersionString] = packageVersion
         plist[PlistKeys.LSApplicationCategoryType] = "Unknown"
-        plist[PlistKeys.CFBundleVersion] = packageVersion
+        val packageBuildVersion = packageBuildVersion.get()!!
```
I think `packageBuildVersion.get()!!` should be replaced with `packageBuildVersion.orNull`, since the property is marked as optional (which is good, as it allows the task to be used separately from the default plugin configuration without bothering with setting `packageBuildVersion`)

Thomas-Vos replied:
Sounds good, I did not know this package task could be used standalone, see latest commit. It now uses `packageBuildVersion.orNull` and has `packageVersion` as a fallback (for backwards compatibility and to make packageBuildVersion really optional). What do you think about this?

AlexeyTsvetkov reviewed (commented):
In general the PR looks good to me.

The docs could have some more details 
on when to use this new property,
@Thomas-Vos what do you think?

Thomas-Vos commented:
Updated the PR with a fix in `AbstractJPackageTask`, added some more documentation about the build version, and fixed some typos in the existing documentation.

AlexeyTsvetkov commented:
@Thomas-Vos thanks for the contribution!
